operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -79,14 +79,14 @@ public Controller(Reconciler<P> reconciler,
     if (size == 0) {
       dependents = Collections.emptyMap();
     } else {
-      final var dependentsHolder = new HashMap<String, DependentResource>(size);
-      specs.forEach((name, drs) -> {
+      final Map<String, DependentResource> dependentsHolder = new LinkedHashMap<>(size);
+      specs.forEach(drs -> {
         final var dependent = createAndConfigureFrom(drs, kubernetesClient);
         // check if dependent implements Deleter to record that fact
         if (!hasDeleterHolder[0] && dependent instanceof Deleter) {
           hasDeleterHolder[0] = true;
         }
-        dependentsHolder.put(name, dependent);
+        dependentsHolder.put(drs.getName(), dependent);
       });
       dependents = Collections.unmodifiableMap(dependentsHolder);
```

Comment:
If we don't want to use list but an ordered Map, thus LinkedHashMap, we should have it also on the defined variable on line 57, so it's explicit on the definition.   (For some reason I cannot comment on that line)

Reply:
The problem if we do that is that we cannot use `Collections.emptyMap` or `Collections.unmodifiableMap` anymore… Then again, since the dependents are not exposed, it's probably OK and indeed conveys the intent more clearly.